wagtail/core/migrations/0001_squashed_0016_change_page_url_path_to_text_field.py

```diff
@@ -41,13 +41,6 @@ def initial_data(apps, schema_editor):
         url_path='/home/',
     )
 
-    # Create default site
-    Site.objects.get_or_create(
-        hostname='localhost',
-        root_page_id=homepage.id,
-        is_default_site=True
-    )
```

Comment:
Scratch that! I see this now happens in https://github.com/wagtail/wagtail/pull/5457/files#diff-5918a29076d7dc71b2122144560a898eR28

Reply:
We need to take extreme care with this - making retrospective changes to existing migrations has the potential to break existing sites in ways that are very hard to put right afterwards.

Reply:
Agreed - Changing prior migrations is almost never a good idea. I'm hoping this is the exception :)

My goal in changing this migration like this was to allow the end user's site table to be created prior to the default data creation, but still allow default data to be injected in. The issues that I think need to be considered are: 

1.  That the new site migration will run for everyone, even if you already ran this migration 
1a. @ababic identified a potential issue with that that I need to address.
2.  Are the changes to old migrations changing database structure? 
2a. In this case there are no database structure changes - just initial data
3. What would a upgrade followed by a rollback look like
3a. The rollback wouldn't delete the initial data that used to be created by this migration. I think that doesn't matter since the table would be dropped.


We're sort of lucky in that it's basically impossible to switch to a swappable model midway through a project, which means the only usecases I could see needing to be tested are:

1. Existing project not using the new functionality 
2. New project not using new functionality
3. New project using new functionality

with no overlap between them.

I think this is the most risky part of the PR so if there are any considerations that others can think of please chime in.

Reply:
Just a thought... Is splitting this into two migrations necessary? As much as it's nice to separate concerns, Site is a core app model, and I think swappable should allow us to access a custom site model here?

Reply:
My understanding is that the order must go:
1. Registration of the swappable model  (here in the squashed migration)
2. Creation of the model table (either here or in the user's initial migration)
3. Writing to the model table (in sites 0001)

I think swappable would allow us to access the custom site model here, but I don't think the table would be created yet as step 2 would have a dependency on this migration.